[BUG] unresolved external symbol std::codecvt for MSVS 2017 with OIIO 2.4.7.1

**Describe the bug**
I'm currently trying to update the OIIO Conan package to 2.4.7.1. During that process, I noticed that OIIO 2.4.7.1 does not build with MSVS 2017 due to an unresolved external symbol.

**To Reproduce**
Steps to reproduce the behavior:
1. Build OIIO 2.4.7.1 with MSVS 2017
2. The build stops due to an unresolved external symbol

Error message from Conan CI run, where the error occured (abbreviated, Full log can be found [here](https://c3i.jfrog.io/c3i/misc/logs/pr/15254/12-configs/windows-visual_studio/openimageio/2.4.7.1//c971b0abf6c185a27b419f1b3a804ccf92f75801-build.txt)):
```
     Creating library C:/J/w/prod/BuildSingleReference@6/s/3dcd59/1/build_subfolder/lib/OpenImageIO_Util.lib and object C:/J/w/prod/BuildSingleReference@6/s/3dcd59/1/build_subfolder/lib/OpenImageIO_Util.exp
strutil.obj : error LNK2019: unresolved external symbol "__declspec(dllimport) public: static class std::locale::id std::codecvt<char16_t,char,struct _Mbstatet>::id" (__imp_?id@?$codecvt@_SDU_Mbstatet@@@std@@2V0locale@2@A) referenced in function "public: __cdecl std::wstring_convert<class std::codecvt_utf8_utf16<char16_t,1114111,0>,char16_t,class std::allocator<char16_t>,class std::allocator<char> >::wstring_convert<class std::codecvt_utf8_utf16<char16_t,1114111,0>,char16_t,class std::allocator<char16_t>,class std::allocator<char> >(void)" (??0?$wstring_convert@V?$codecvt_utf8_utf16@_S$0BAPPPP@$0A@@std@@_SV?$allocator@_S@2@V?$allocator@D@2@@std@@QEAA@XZ) [C:\J\w\prod\BuildSingleReference@6\s\3dcd59\1\build_subfolder\source_subfolder\src\libutil\OpenImageIO_Util.vcxproj]
C:\J\w\prod\BuildSingleReference@6\s\3dcd59\1\build_subfolder\bin\OpenImageIO_Util.dll : fatal error LNK1120: 1 unresolved externals [C:\J\w\prod\BuildSingleReference@6\s\3dcd59\1\build_subfolder\source_subfolder\src\libutil\OpenImageIO_Util.vcxproj]
openimageio/2.4.7.1: 
```

**Expected behavior**

OIIO should build with MSVS 2017, since it's in the list of supported compilers.

**Platform information:**
 - OIIO branch/version: 2.4.7.1
 - OS: Windows 
 - C++ compiler: MSVS 2017
 - Any non-default build flags when you build OIIO:

**Investigation**

In `src/libutil/strutil.cpp:929`, `char16_t` is passed as a template parameter:
```c++
std::string
Strutil::utf16_to_utf8(const std::u16string& str) noexcept
{
    try {
        OIIO_PRAGMA_WARNING_PUSH
        OIIO_CLANG_PRAGMA(GCC diagnostic ignored "-Wdeprecated-declarations")
        std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> conv;
        OIIO_PRAGMA_WARNING_POP
        return conv.to_bytes(str);
    } catch (const std::exception&) {
        return std::string();
    }
}
```
This results in the undefined symbol. It's a bug in MSVS 2017, see [here](https://stackoverflow.com/questions/32055357/visual-studio-c-2015-stdcodecvt-with-char16-t-or-char32-t/35103224#35103224) on Stackoverflow and [here](https://social.msdn.microsoft.com/Forums/en-US/8f40dcd8-c67f-4eba-9134-a19b9178e481/vs-2015-rc-linker-stdcodecvt-error?forum=vcgeneral) on MSDN

**Workaround/possible fix:**

I found that vcpkg has encountered a similar issue (https://github.com/microsoft/vcpkg/issues/28735) and they fix it with this patch:
```
diff --git a/src/libutil/strutil.cpp b/src/libutil/strutil.cpp
index 129a9b5..09baea8 100644
--- a/src/libutil/strutil.cpp
+++ b/src/libutil/strutil.cpp
@@ -914,8 +914,15 @@ std::string
 Strutil::utf16_to_utf8(const std::u16string& str) noexcept
 {
     try {
+        // https://stackoverflow.com/a/35103224
+#if defined _MSC_VER && _MSC_VER >= 1900 && _MSC_VER < 1930
+        std::wstring_convert<std::codecvt_utf8_utf16<int16_t>, int16_t> convert;
+        auto p = reinterpret_cast<const int16_t *>(str.data());
+        return convert.to_bytes(p, p + str.size());
+#else
         std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> conv;
         return conv.to_bytes(str);
+#endif
     } catch (const std::exception&) {
         return std::string();
     }
```
Unfortunatly, I don't have a MSVS 2017 installation locally, but I tried this fix in the Conan CI pipeline and now it builds correctly.

I'm not sure, if this is the best way to fix this, so I didn't file a PR

--- a/src/libutil/strutil.cpp
+++ b/src/libutil/strutil.cpp
@@ -929,7 +929,7 @@ Strutil::utf16_to_utf8(const std::u16string& str) noexcept
         // There is a bug in MSVS 2017 causing an unresolved symbol if char16_t is used (see https://stackoverflow.com/a/35103224)
 #if defined _MSC_VER && _MSC_VER >= 1900 && _MSC_VER < 1930
         std::wstring_convert<std::codecvt_utf8_utf16<int16_t>, int16_t> convert;
-        auto p = reinterpret_cast<const int16_t *>(str.data());
+        auto p = reinterpret_cast<const int16_t*>(str.data());
         return convert.to_bytes(p, p + str.size());
 #else
         std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> conv;

--- a/src/libutil/strutil.cpp
+++ b/src/libutil/strutil.cpp
@@ -926,9 +926,16 @@ Strutil::utf16_to_utf8(const std::u16string& str) noexcept
     try {
         OIIO_PRAGMA_WARNING_PUSH
         OIIO_CLANG_PRAGMA(GCC diagnostic ignored "-Wdeprecated-declarations")
+        // There is a bug in MSVS 2017 causing an unresolved symbol if char16_t is used (see https://stackoverflow.com/a/35103224)
+#if defined _MSC_VER && _MSC_VER >= 1900 && _MSC_VER < 1930
+        std::wstring_convert<std::codecvt_utf8_utf16<int16_t>, int16_t> convert;
+        auto p = reinterpret_cast<const int16_t *>(str.data());
+        return convert.to_bytes(p, p + str.size());
+#else
         std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> conv;
-        OIIO_PRAGMA_WARNING_POP
         return conv.to_bytes(str);
+#endif
+        OIIO_PRAGMA_WARNING_POP
     } catch (const std::exception&) {
         return std::string();
     }

